File: UserRegistration/Services/UserServices/IUserServices.cs
using UserRegistration.Models;
using UserRegistration.Dtos.User;

namespace UserRegistration.Services.UserServices
{
    public interface IUserServices
    {
        public Task<List<GetUserDto>> GetUsers();
        public Task<GetUserDto> GetSingleUser(string id);
        public Task<bool> AddUser(AddUpdateUserDto user);
        public Task<bool> UpdateUser(string id, AddUpdateUserDto request);
        public Task<bool> DeleteUser(string id);
    }
}

File: web/src/services/iUserServices.php
<?php

namespace Web\services;

use Web\models\User;

interface iUserServices
{
    public function get_users(): array;
    public function get_user_by_id($user_id): User;
    public function add_user(User $user): bool;
    public function update_user(User $user): bool;
    public function delete_user($user_id): bool;
}
